### PR TITLE
Fix action run on base branch

### LINKED ISSUE #27854: [Bug]: GitHub Action on PR of old base would fail
### What happened?

Example PR: https://github.com/apache/beam/pull/27853 

It has a dumb change based on an old master (sometimes in July), its GitHub Action fails:

https://github.com/apache/beam/actions/runs/5764938541/job/15629910028?pr=27853

We should run the Actions on the branch of (Change merge into master), not the PR branch directly

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #27854

example run:
- pull request creation: https://github.com/Abacn/beam/actions/runs/5766352092/job/15634107987 in https://github.com/Abacn/beam/pull/57
- pull request phrase trigger: https://github.com/Abacn/beam/actions/runs/5766319108/job/15634024620 in https://github.com/Abacn/beam/pull/56

it appears there is still an issue in the pull request trigger. Once triggered, it will abort the push or schedule run on master branch

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
R: @andreydevyatkin @damccorm

**Comment by Abacn** [comment]
It fails on this PR due to inconsistent setup-action.yml run between setup (yml on master) and post setup (yml been overwritten). This is actually the same direct cause of #27854 a github action issue

File: .github/actions/setup-action/action.yml [hunk 1]
```diff
@@ -36,16 +36,20 @@ runs:
       uses: actions/checkout@v3
       if: |
         github.event_name == 'push' ||
-        github.event_name == 'pull_request_target' ||
         github.event_name == 'schedule' ||
         github.event_name == 'workflow_dispatch'
       with:
         ref: ${{ github.event.pull_request.head.sha }}
```
Comment (damccorm) [inline .github/actions/setup-action/action.yml:42]: Hm, for push/schedule/workflow_dispatch, we probably don't even need this checkout, do we?

Comment (damccorm) [reply .github/actions/setup-action/action.yml:42]: Using the pull_request event doesn't really make sense and we've already done the checkout previously to get access to this action

Comment (Abacn) [reply .github/actions/setup-action/action.yml]: yeah that's right. Changed and tested on https://github.com/Abacn/beam/actions/runs/5787197443/job/15683590845

**Review by damccorm — APPROVED** [review]
LGTM

**Comment by damccorm** [comment]
Looks like this didn't work - https://github.com/apache/beam/actions/runs/5795001413/job/15705720400?pr=27893

**Pull request merged**